distribution/test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSingleSignOn.java

```diff
@@ -0,0 +1,261 @@
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.core.AllOf.allOf;
+import static org.hamcrest.core.CombinableMatcher.both;
+import static org.mockito.AdditionalMatchers.not;
+import static com.jayway.restassured.RestAssured.get;
+import static com.jayway.restassured.RestAssured.given;
+
+import java.io.StringReader;
+import java.net.URI;
+import java.net.URL;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.XMLConstants;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.codice.ddf.security.common.jaxrs.RestSecurity;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.osgi.service.cm.Configuration;
+
+import com.jayway.restassured.response.Response;
+
+import ddf.common.test.BeforeExam;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class TestSingleSignOn extends AbstractIntegrationTest {
+
+    public static final String IDP_AUTH_TYPES = "/=SAML|ANON,/search=SAML|IDP,/solr=SAML|PKI|basic";
+
+    private static final DynamicUrl SEARCH_URL = new DynamicUrl(DynamicUrl.SECURE_ROOT, HTTPS_PORT,
+            "/search");
+
+    private static final DynamicUrl WHO_AM_I_URL = new DynamicUrl(SERVICE_ROOT, "/whoami");
+
+    @BeforeExam
+    public void beforeTest() throws Exception {
+        basePort = getBasePort();
+        getAdminConfig().setLogLevels();
+        getSecurityPolicy().configureWebContextPolicy(null, IDP_AUTH_TYPES, null, null);
+        getServiceManager().waitForAllBundles();
+        getServiceManager().waitForHttpEndpoint(SERVICE_ROOT + "/catalog/query");
+        getServiceManager().waitForHttpEndpoint(WHO_AM_I_URL.getUrl());
+
+        // Start the services needed for testing. We need to start the Search UI to test that it redirects properly
+        getServiceManager().startFeature(true, "security-idp", "search-ui-app");
+
+        // Create a validator from the metadata schema so that we can validate the metadata
+        // TODO: Maybe don't go out to the internet for this? The schema itself is going to go out to the internet anyway though.
+        URL schemaFile = new URL(
+                "http://docs.oasis-open.org/security/saml/v2.0/saml-schema-metadata-2.0.xsd");
```

Comment:
That one XSD depends on several others, should I put them all in the resources directory?